nextcloud/server
# Make it possible to filter by tags with REPORT method

From https://github.com/owncloud/core/pull/26099

Enhanced the REPORT method on the Webdav endpoint and added a
"oc:favorite" filter rule. When set, it will return a flat list of
results filtered with only favorite files.

The web UI was also adjusted to use this REPORT method instead of the
private API endpoint.

CC: @nickvergessen @MorrisJobke @icewind1991 @LukasReschke 

Test steps:
1. Favorite a few files in folders and subfolders, share one of them with link (for the share indicator).
2. Go to the "Favorites" section in the web UI.
3. Check the network console and see that it's using REPORT to get the results.
4. Check that the displayed results look fine and also contain the share indicator from the previous link share (it's part of the Webdav response)
